lib/mix/tasks/compile/live_view.ex

```diff
@@ -0,0 +1,99 @@
+defmodule Mix.Tasks.Compile.LiveView do
+  use Mix.Task
+  @recursive true
+
+  alias Mix.Task.Compiler.Diagnostic
+
+  @doc false
+  def run(args) do
+    {compile_opts, _argv, _errors} = OptionParser.parse(args, strict: [return_errors: :boolean])
+
+    case validate_components_calls(project_modules()) do
```

Comment:
So I think we should add a very basic manifest, otherwise this will definitely be too expensive. Here is the idea:

1. You will store all diagnostics in a file, like this `{1, diagnostics}`, where 1 is used for versioning. You can write this file to disk using `:erlang.term_to_binary`

2. Now you will compare your manifest file with Mix.Tasks.Compile.Elixir.manifests(). If your manifest is missing or it is older than any of the elixir manifests, then you recompute the diagnostics

3. If your manifest is not older and the `--all-warnings` flag is given, then you will return `{:noop, diagnostics_from_manifest}`. Feel free to ping me if you want to go over this in a chat. :)

Reply:
@josevalim done! Additionally, I also had to handle `--warnings-as-errors` and `--force`. 

Addressing the other smaller suggestions now.